test-adapter/src/main/scala/org/scalajs/testing/adapter/JSEnvRPC.scala

```diff
@@ -15,14 +15,35 @@ package org.scalajs.testing.adapter
 import scala.concurrent.ExecutionContext
 
 import org.scalajs.jsenv._
+import org.scalajs.logging.Logger
 import org.scalajs.testing.common._
 
 /** RPC Core for use with a [[JSEnv]]. */
 private[adapter] final class JSEnvRPC(
-    jsenv: JSEnv, input: Seq[Input], config: RunConfig)(
+    jsenv: JSEnv, input: Seq[Input], logger: Logger)(
     implicit ec: ExecutionContext) extends RPCCore {
 
-  private val run = jsenv.startWithCom(input, config, handleMessage)
+  private val run: JSComRun = {
+    /* #4560 Explicitly redirect out/err to System.out/System.err, instead of
+     * relying on `inheritOut` and `inheritErr`, so that streams installed with
+     * `System.setOut` and `System.setErr` are always taken into account.
+     * sbt installs such alternative outputs when it runs in server mode.
+     *
+     * We never wait for these threads to finish. In theory, tasks that use the
+     * test adapter may complete before all output has been transferred to
+     * `System.out` and `System.err`.
```

Comment:
While that is fair from an execution schedule POV, it may swallow errors. Probably calling `reportFailure` as defined on the `TestAdapter` is a good stop-gap solution (maybe wrap with a different error).

Reply:
Like for `run`, I don't see this as a real issue. If piping of the stdout/stderr streams fails, there will be an error printed on stderr. That seems enough to me (especially given the very low likelihood that any error *would* happen in there that would not also involve a failure of the process and hence an exception being thrown through a different channel).

Reply:
Similar argument: For human/interactive use, this is absolutely fine. However, for CI or other automated use, such behavior is very hostile.